src/libraries/System.Runtime/tests/System/AppContext/SecureAppContext.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using Microsoft.DotNet.RemoteExecutor;
+using Xunit;
+using Xunit.Sdk;
+
+namespace System.Tests
+{
+    public partial class SecureAppContextTests
+    {
+        // these tests only make sense on platforms where reflection is expected to forbid overwriting initonly fields
+        public static bool RunForbidSettingInitonlyTests => PlatformDetection.IsNetCore && RemoteExecutor.IsSupported;
+
+        [ConditionalFact(nameof(RunForbidSettingInitonlyTests))]
+        public void CannotUseReflectionToChangeValues()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                Type secureAppContextType = typeof(AppContext).Assembly.GetType("System.SecureAppContext", throwOnError: true);
+
+                foreach (FieldInfo field in secureAppContextType.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static))
+                {
+                    try
+                    {
+                        Assert.True(field.FieldType.IsValueType, "Field is a reference type; instance members may be subject to mutation.");
+                        Assert.True(field.IsInitOnly, "Field is mutable.");
```

Comment:
These messages are wrong for xUnit.

In the event of failure it'll say something like

```
  Assertion failed: Field is a reference type; instance members may be subject to mutation.
    Expected: True
        Actual: False
```

or

```
  Assertion failed: Field is mutable.
    Expected: True
        Actual: False
```

Since the assertion message prints what was expected, they should essentially restate the condition expression.

```suggestion
                        Assert.True(field.FieldType.IsValueType, "Field is a value type.");
                        Assert.True(field.IsInitOnly, "Field is declared readonly.");
```

Reply:
This code is now deleted.